### ISSUE TITLE
Unify the evaluation metric such as accuracy API with the operators

### ISSUE THREAD

**Issue by futurely** [dmlc/mxnet#661]
Treat evaluation operations specially with a API different from the operators add unnecessary confusion and complexity in model definition. 
Where to plug in the accuracy?
```python
softmax = mx.symbol.SoftmaxOutput(data=fc, name="softmax")
return softmax
```
Appearing along with its input is a much clearer solution.
```python
softmax = mx.symbol.SoftmaxOutput(data=fc, name="softmax")
accuracy = mx.symbol.Accuracy(softmax, name = 'accuracy')
return accuracy
```

**Comment by jermainewang** [comment]
Hi,

Thank you for you suggestion. With your code, you need to do softmax in order to compute accuracy. But this is a redundant computation, since one could directly compute accuracy without softmax. Maybe return some thing like:
```python
softmax = mx.symbol.SoftmaxOutput(data=fc, name="softmax")
accuracy = mx.symbol.Accuracy(data=fc, name = "accuracy")
return softmax, accuracy
```
is better?

-Minjie

**Comment by futurely** [comment]
You're right. Softmax is only for normalized class probabilities. The update signatures of `Accuracy` in the two cases are the same.

**Comment by tqchen** [comment]
This specification really depends on what was wanted. I will mark this a discussion thread. 

From some perspective, it is cleaner from a modeling point of view to separate evaluation, model configuration and loss function(which we did not). The idea is model configurations contains everything that is necessary for prediction, but not others. Some comparisons

- The current evaluation interface works nicely for users to quickly re-define new metrics
- The current interface allows users to change evaluation metric easily to re-evaluate a trained net
- The current interface is not very friendly for multiple output and loss functions.

The solution can be either what you proposed, or some improved evaluation metric interface that handles multiple outputs gracefully

**Comment by futurely** [comment]
The proposed interface enables even easier replacement of `accuracy = mx.symbol.Accuracy(data=fc, name = "accuracy")` with `euclidean = mx.symbol.Euclidean(data=fc, name = "regression")`.

Re-evaluation is still possible as long as the prediction is the output. `bboxes, pascal_voc_mean_average_precision = get_symbols()` and then `average_precision = mx.symbol.MsCocoAveragePrecision(bboxes)`.

Multiple outputs are not so different from single output conceptually. [Here](https://github.com/rbgirshick/py-faster-rcnn/issues/16#issuecomment-155055447) is an concrete example of combing layers with multiple outputs and multiple inputs.
